src/wp-includes/class-wp-duotone.php

```diff
@@ -0,0 +1,1064 @@
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * @package WordPress
+ * @since 6.3.0
+ */
+
+/**
+ * Manages duotone block supports and global styles.
+ *
+ * @access public
+ */
+class WP_Duotone {
+	/**
+	 * Block names from global, theme, and custom styles that use duotone presets and the slug of
+	 * the preset they are using.
+	 *
+	 * Example:
+	 *  [
+	 *      'core/featured-image' => 'blue-orange',
+	 *       …
+	 *  ]
+	 *
+	 * @var array
+	 */
+	private static $global_styles_block_names = array();
+
+	/**
+	 * An array of duotone filter data from global, theme, and custom presets.
+	 *
+	 * Example:
+	 *  [
+	 *      'wp-duotone-blue-orange' => [
+	 *          'slug'  => 'blue-orange',
+	 *          'colors' => [ '#0000ff', '#ffcc00' ],
+	 *      ],
+	 *      'wp-duotone-red-yellow' => [
+	 *          'slug'   => 'red-yellow',
+	 *          'colors' => [ '#cc0000', '#ffff33' ],
+	 *      ],
+	 *      …
+	 *  ]
+	 *
+	 * @var array
+	 */
+	private static $global_styles_presets = array();
+
+	/**
+	 * All of the duotone filter data from presets for CSS custom properties on
+	 * the page.
+	 *
+	 * Example:
+	 *  [
+	 *      'wp-duotone-blue-orange' => [
+	 *          'slug'   => 'blue-orange',
+	 *          'colors' => [ '#0000ff', '#ffcc00' ],
+	 *      ],
+	 *      …
+	 *  ]
+	 *
+	 * @var array
+	 */
+	private static $used_global_styles_presets = array();
+
+	/**
+	 * All of the duotone filter data for SVGs on the page. Includes both
+	 * presets and custom filters.
+	 *
+	 * Example:
+	 *  [
+	 *      'wp-duotone-blue-orange' => [
+	 *          'slug'   => 'blue-orange',
+	 *          'colors' => [ '#0000ff', '#ffcc00' ],
+	 *      ],
+	 *      'wp-duotone-000000-ffffff-2' => [
+	 *          'slug'   => '000000-ffffff-2',
+	 *          'colors' => [ '#000000', '#ffffff' ],
+	 *      ],
+	 *      …
+	 *  ]
+	 *
+	 * @var array
+	 */
+	private static $used_svg_filter_data = array();
+
+	/**
+	 * All of the block CSS declarations for styles on the page.
+	 *
+	 * Example:
+	 *  [
+	 *      [
+	 *          'selector'     => '.wp-duotone-000000-ffffff-2.wp-block-image img',
+	 *          'declarations' => [
+	 *              'filter' => 'url(#wp-duotone-000000-ffffff-2)',
+	 *          ],
+	 *      ],
+	 *      …
+	 *  ]
+	 *
+	 * @var array
+	 */
+	private static $block_css_declarations = array();
+
+	/**
+	 * Direct port of colord's clamp function. Using min/max instead of
+	 * nested ternaries.
+	 *
+	 * @see https://github.com/omgovich/colord/blob/3f859e03b0ca622eb15480f611371a0f15c9427f/src/helpers.ts#L23
+	 *
+	 * @param float $number The number to clamp.
+	 * @param float $min   The minimum value.
+	 * @param float $max   The maximum value.
+	 * @return float The clamped value.
+	 */
+	private static function colord_clamp( $number, $min = 0, $max = 1 ) {
+		return $number > $max ? $max : ( $number > $min ? $number : $min );
+	}
+
+	/**
+	 * Direct port of colord's clampHue function.
```

Comment:
```suggestion
	 * Processes and clamps a degree (angle) value properly.
```

This applies throughout -- the docblocks are used to generate developer.wordpress.org so need to explain what the method does.

Reply:
These methods are all private. Is there a way to not generate documentation for this on developer.wordpress.org?